components/entities/api/src/main/java/org/phenotips/entities/PrimaryEntity.java

```diff
@@ -53,7 +55,14 @@
      *
      * @return a valid document reference
      */
-    DocumentReference getDocument();
+    DocumentReference getDocumentReference();
+
+    /**
+     * Returns the document where the entity is stored.
+     *
+     * @return a valid document reference
+     */
+    XWikiDocument getDocument();
```

Comment:
Considering that the upgrade would break things anyway, and considering that changing the API would make it easier to understand and use in the future, I'm ok with the return type of `getDocument()` being changed to `XWikiDocument`. So there doesn't need to be any change to this PR.